raphtory/src/python/graph/properties/temporal_props.rs

```diff
@@ -244,23 +243,17 @@ impl PyTemporalProp {
     }
 
     /// List of ordered deduplicated property values
-    pub fn ordered_dedupe(&self, latest_time: bool) -> Vec<(i64, Prop)> {
+    pub fn ordered_dedupe(&self, latest_time: bool) -> Vec<(TimeIndexEntry, Prop)> {
         self.prop.ordered_dedupe(latest_time)
     }
 
-    /// List update timestamps and corresponding property values
-    pub fn items_date_time(&self) -> Option<Vec<(DateTime<Utc>, Prop)>> {
-        Some(self.prop.histories_date_time()?.collect())
-    }
-
     /// Iterate over `items`
     pub fn __iter__(&self) -> PyBorrowingIterator {
-        py_borrowing_iter!(self.prop.clone(), DynTemporalProperty, |inner| inner
-            .histories())
+        py_borrowing_iter!(self.prop.clone(), DynTemporalProperty, |inner| inner.iter())
     }
     /// Get the value of the property at time `t`
-    pub fn at(&self, t: PyTime) -> Option<Prop> {
-        self.prop.at(t.into_time())
+    pub fn at(&self, t: TimeIndexEntry) -> Option<Prop> {
```

Comment:
this should probably use the same handling as time inputs for additions. If a secondary index is given explicitly it should be used, otherwise it should give you the last value with that timestamp